mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -479,6 +500,25 @@ object KMeans {
   }
 
   /**
+   * Trains a k-means model using the given set of parameters and initial cluster centers
+   *
+   * @param data training points stored as `RDD[Vector]`
+   * @param k number of clusters
+   * @param maxIterations max number of iterations
+   * @param initialModel an existing set of cluster centers.
+   */
+  def train(
```

Comment:
I agree.  This extra static method is not necessary since we decided we prefer the builder pattern, as @srowen said.